shishito/runtime/environment/browserstack.py

```diff
@@ -63,54 +63,91 @@ def get_pytest_arguments(self, config_section):
+                '--platform=': '--platform=' + platform,
+                '--device=': '--device=' + device,
+                '--deviceOrientation=': '--deviceOrientation=' + deviceOrientation,
+                '--browserstack=': '--browserstack=' + bs_auth
+            }
 
     def get_capabilities(self, config_section):
         """ Return dictionary of capabilities for specific config combination.
 
         :param str config_section: section in platform/environment.properties config
         :return: dict with capabilities
         """
-
+        test_platform = self.shishito_support.test_platform
         get_opt = self.shishito_support.get_opt
-
-        return {
-            'acceptSslCerts': get_opt('accept_ssl_cert').lower() == 'false',
-            'browserstack.debug': get_opt('browserstack_debug').lower(),
-            'project': get_opt('project_name'),
-            'build': get_opt('build_name'),
-            'os': get_opt(config_section, 'os'),
-            'os_version': get_opt(config_section, 'os_version'),
-            'browser': get_opt(config_section, 'browser'),
-            'browser_version': get_opt(config_section, 'browser_version'),
-            'resolution': get_opt(config_section, 'resolution'),
-            'name': self.get_test_name() + time.strftime('_%Y-%m-%d'),
-            'browserstack.local': get_opt('browserstack_local') or False,
+        if(test_platform == 'web'):
+            return {
+                'acceptSslCerts': get_opt('accept_ssl_cert').lower() == 'false',
+                'browserstack.debug': get_opt('browserstack_debug').lower(),
+                'project': get_opt('project_name'),
+                'build': get_opt('build_name'),
+                'os': get_opt(config_section, 'os'),
+                'os_version': get_opt(config_section, 'os_version'),
+                'browser': get_opt(config_section, 'browser'),
+                'browser_version': get_opt(config_section, 'browser_version'),
+                'resolution': get_opt(config_section, 'resolution'),
+                'name': self.get_test_name() + time.strftime('_%Y-%m-%d'),
+                'browserstack.local': get_opt('browserstack_local') or False,
         }
+        if(test_platform == 'mobile'):
+            return {
+                'acceptSslCerts': get_opt('accept_ssl_cert').lower() == 'false',
+                'browserstack.debug': get_opt('browserstack_debug').lower(),
+                'project': get_opt('project_name'),
+                'build': get_opt('build_name'),
+                'browser': get_opt(config_section, 'browser'),
+                'platform': get_opt(config_section, 'platform'),
+                'device': get_opt(config_section, 'device'),
+                'deviceOrientation': get_opt(config_section, 'deviceOrientation') or 'portrait',
+
+                'name': self.get_test_name() + time.strftime('_%Y-%m-%d'),
+                'browserstack.local': get_opt('browserstack_local') or False,
+            }
```

Comment:
There is a lot of identical return parameters. May be worth extracting the parameters that are common for both web and mobile, and in the if/else just update the specific ones